node-wasm/src/worker.rs

```diff
@@ -48,12 +41,88 @@ pub enum WorkerError {
     NodeError(Error),
 }
 
+#[wasm_bindgen]
 struct NodeWorker {
+    event_channel_name: String,
+    worker: Mutex<Option<NodeWorkerInstance>>,
+    request_server: Mutex<RequestServer>,
+    control_channel: mpsc::Sender<JsValue>,
+}
+
+struct NodeWorkerInstance {
     node: Node<IndexedDbStore>,
     events_channel_name: String,
 }
 
+#[wasm_bindgen]
 impl NodeWorker {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        info!("Created lumina worker");
+
+        let (tx, rx) = mpsc::channel(NODE_WORKER_QUEUE_SIZE);
+
+        Self {
+            event_channel_name: format!("NodeEventChannel-{}", random_id()),
+            worker: Mutex::new(None),
+            request_server: Mutex::new(RequestServer::new(rx)),
+            control_channel: tx,
+        }
+    }
+
+    pub async fn connect(&self, port: JsValue) {
+        self.control_channel
+            .send(port)
+            .await
+            .expect("RequestServer command channel should never close")
+    }
+
+    #[wasm_bindgen(js_name = runWorker)]
+    pub async fn run_worker(&self) -> Result<(), Error> {
```

Comment:
You can use `&mut self` here and remove mutexes.

Reply:
Originally I've added this to avoid runtime errors about js violating rust borrow rules (since `NodeWorker` would be owned by js). I expected `run_workers` and `connect` being called from callback to clash, but I can't seem to trigger it with current implementation based on channels. 
So switching to mut as suggested :+1: